src/Consumption/QueueExtension.php

```diff
@@ -132,8 +133,8 @@ public function onPostMessageReceived(PostMessageReceived $context): void
      */
     public function onPreConsume(PreConsume $context): void
     {
-        $this->runtime = microtime(true) - $this->startedAt;
-        $this->logger->debug(sprintf('Runtime: %s', $this->runtime));
+        $this->runtime = $this->startedAt->diffInSeconds();
+        $this->logger->debug(sprintf('Runtime (seconds): %d', $this->runtime));
```

Comment:
I think we should retain the millisecond precision here. It could be logged as `Runtime (seconds): 12312.1231`. But there will be tasks that have less than 1 second of runtime and we don't want to log `0`.

Reply:
That does seem beneficial. I'm not seeing any obvious way to get a diff in milliseconds using Chronos though. Any pointers?

Reply:
I did some fancy stuff with DateIntervals. I think it should work pretty well.

I'm starting to wonder if the QueueExtension should be in it's own plugin. We should probably have a way for the user to create their own plugins and then load them all when the worker starts.

Reply:
Couldn't you do something like

```php
$duration = ($end->getTimestamp() + $end->microseconds / 10000) - ($start->getTimestamp() + $start->microseconds / 10000);
```

It's not the most elegant solution, but it'll get the job done.